sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +26,245 @@ class StaticPlaylist
+
+    /**
+     *  Returns a read-only (constant) reverse iterator that points
+     *  to one before the first element in the %vector.  Iteration
+     *  is done in reverse element order.
+     */
+    const_reverse_iterator
+    rend() const _GLIBCXX_NOEXCEPT
+    { return const_reverse_iterator(begin()); }
+
+#if __cplusplus >= 201103L
+    /**
+     *  Returns a read-only (constant) iterator that points to the
+     *  first element in the %vector.  Iteration is done in ordinary
+     *  element order.
+     */
+    const_iterator
+    cbegin() const noexcept
+    { return const_iterator(this->m_tracklist.cbegin()); }
+
+    /**
+     *  Returns a read-only (constant) iterator that points one past
+     *  the last element in the %vector.  Iteration is done in
+     *  ordinary element order.
+     */
+    const_iterator
+    cend() const noexcept
+    { return const_iterator(this->m_tracklist.cend()); }
+
+    /**
+     *  Returns a read-only (constant) reverse iterator that points
+     *  to the last element in the %vector.  Iteration is done in
+     *  reverse element order.
+     */
+    const_reverse_iterator
+    crbegin() const noexcept
+    { return const_reverse_iterator(end()); }
+
+    /**
+     *  Returns a read-only (constant) reverse iterator that points
+     *  to one before the first element in the %vector.  Iteration
+     *  is done in reverse element order.
+     */
+    const_reverse_iterator
+    crend() const noexcept
+    { return const_reverse_iterator(begin()); }
+#endif
 
     /** @todo Constructor from any reversible sequence container */
+    /**
+     *  @brief  Creates a %vector with no elements.
+     */
+    StaticPlaylist()
+    {
+        m_tracklist();
+    }
+
+    template<class T>
+    StaticPlaylist(std::initializer_list<T>& __l)
+    {
+        for(const T& i: __l) {
+            m_tracklist.push_back(i);
+        }
+    }
+
+    /**
+     *  @brief  %StaticPlaylist copy constructor.
+     *  @param  __x  A %StaticPlaylist of identical element.
+     */
+    template<class T>
+    StaticPlaylist(const T& __x)
+    {
+        for(typename T::const_iterator i = __x.begin(); i != __x.end(); ++i) {
+            m_tracklist.push_back(*i);
+        }
+    }
+
+#if __cplusplus >= 201103L
+    /**
+     *  @brief  %StaticPlaylist move constructor.
+     */
+    StaticPlaylist(Container&& __x)
+    : m_tracklist(std::forward(__x)) { }
+
+    /**
+     *  @brief  Builds a %StaticPlaylist from an initializer list.
+     *  @param  __l  An initializer_list.
+     */
+#endif
+
+    /**
+     *  @brief  Builds a %StaticPlaylist from a range.
+     *  @param  __first  An input iterator.
+     *  @param  __last  An input iterator.
+     */
+#if __cplusplus >= 201103L
+    template<typename _InputIterator,
+         typename = std::_RequireInputIter<_InputIterator>>
+    StaticPlaylist(_InputIterator __first, _InputIterator __last)
+    {
+        m_tracklist(__first, __last);
```

Comment:
use initialization list

Reply:
corrected. why? is it more optimal call? we have both constructors for the containers